llvm/include/llvm/MC/CAS/MCCASObjectV1.h

```diff
@@ -485,6 +508,73 @@ class MCCASReader {
   const MCSchema &Schema;
 };
 
+/// A DWARFObject implementation that can support dwarfdumping as debug info
+/// compile unit.
+class InMemoryCASDWARFObject : public DWARFObject {
+  ArrayRef<char> DebugAbbrevSection;
+  bool IsLittleEndian;
+
+public:
+  InMemoryCASDWARFObject(ArrayRef<char> AbbrevContents, bool IsLittleEndian)
+      : DebugAbbrevSection(AbbrevContents), IsLittleEndian(IsLittleEndian) {}
+  bool isLittleEndian() const override { return IsLittleEndian; }
+
+  StringRef getAbbrevSection() const override {
+    return toStringRef(DebugAbbrevSection);
+  }
+
+  Optional<RelocAddrEntry> find(const DWARFSection &Sec,
+                                uint64_t Pos) const override {
+    return {};
+  }
+
+  struct PartitionedDebugInfoSection {
+    SmallVector<char> DebugInfoRefData;
+    SmallVector<char> FixupData;
```

Comment:
Here's a maybe crazy idea:

We could make `std::unordered_set<llvm::dwarf::Form> FormsToFixup` a static member of this class.
More importantly, just make this a `constexpr std::array`:

```
constexpr static std::array FormsToPartition{strp, ...};
```

Most functions that are passing this set around as parameters already deal with the class above, so IMO we can get rid of them.
Also, we don't expect this set to be configurable at runtime anyway.
Finally, we can replace the set queries with a `llvm::is_contained(FormsToPartition, SomeForm)`. Yes, it's a linear scan, but this container will never have a lot of elements, so this is likely much faster (plus it can be done at compile time, if `SomeForm` is constexpr)

Reply:
Great suggestion!